fit-checker/utils/change-detect/LectureListDetect.swift

```diff
@@ -0,0 +1,139 @@
+        return newValue.lectures.count - oldValue.lectures.count
+    }
+
+
+
+    /// Detects semester label change.
+    ///
+    /// - Parameters:
+    ///   - oldValue: old value of LectureListResult
+    ///   - newValue: new value of LectureListResult
+    /// - Returns: Change entity or empty if no change detected
+    private func detectSemesterChange(oldValue: LectureListResult, newValue: LectureListResult)
+                    -> Change<String>? {
+        if oldValue.semesterInfo != newValue.semesterInfo {
+            var type: ChangeType
+
+            if oldValue.semesterInfo.isEmpty {
+                type = ChangeType.added
+            } else if newValue.semesterInfo.isEmpty {
+                type = ChangeType.removed
+            } else {
+                type = ChangeType.modified
+            }
+
+            return Change(
+                    type: type,
+                    oldValue: oldValue.semesterInfo,
+                    newValue: newValue.semesterInfo
+            )
+        }
+        return nil
+    }
+
+
+    /// Iterates over all values and returns changes for each index.
+    ///
+    /// - Parameters:
+    ///   - oldValue: old value of LectureListResult
+    ///   - newValue: new value of LectureListResult
+    /// - Returns: Change entity array or empty array if no change detected
+    private func detectValuesChange(oldValue: LectureListResult, newValue: LectureListResult)
+                    -> [Change<Lecture>] {
+        var changes = [Change<Lecture>]()
+        let oldLectures = oldValue.lectures
+        let newLectures = newValue.lectures
+
+        var oldIndex = 0
+        var newIndex = 0
+
+        while oldIndex < oldLectures.count || newIndex < newLectures.count {
+            let oldItem = ArrayUtils<Lecture>.getItemSafely(array: oldLectures, i: oldIndex)
+            let newItem = ArrayUtils<Lecture>.getItemSafely(array: newLectures, i: newIndex)
+
+            let change = detectValueChange(oldItem: oldItem, newItem: newItem)
+            if let change = change {
+                changes.append(change)
+            }
+
+            oldIndex += 1
+            newIndex += 1
+        }
+
+        return changes
+    }
+
+
+
+    /// Detects if change occured. By checking if one of items is not 
+    /// missing. If both are equal, then their name is checked.
+    ///
+    /// - Parameters:
+    ///   - oldValue: old value of Lecture
+    ///   - newValue: new value of Lecture
+    /// - Returns: Change entity or empty if no change detected
+    private func detectValueChange(oldItem: Lecture?, newItem: Lecture?) -> Change<Lecture>? {
+        var type: ChangeType?
+
+        type = (oldItem == nil ? ChangeType.added : nil)
+
+        type = (newItem == nil ? ChangeType.removed : type)
+
+        if let old = oldItem, let new = newItem {
+            type = (old.name != new.name ? ChangeType.modified : type)
```

Comment:
Here you can move the comparison to the `if-let`statement, then you can drop the ternary operator:
```swift
if let ..., let ..., old != new {
    type = .modified
}
```